components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedIdPData.java

```diff
@@ -224,4 +224,9 @@ public Object clone() throws CloneNotSupportedException {
         authenticatedIdPData.authenticators = new ArrayList<>(this.authenticators);
         return authenticatedIdPData;
     }
+
+    public void setAuthenticators(List<AuthenticatorConfig> authenticators) {
+
+        this.authenticators = authenticators;
+    }
```

Comment:
let's revisit the need to add this public methods

Reply:
We are using it in the loading process of the session context.